app/uk/gov/hmrc/helptosavestub/controllers/NSIController.scala

```diff
@@ -98,8 +101,32 @@ object NSIController extends BaseController with Logging {
   def createAccount(): Action[AnyContent] = Action { implicit request ⇒
     val description = "create account"
     withNSIPayload(description){ nsiPayload ⇒
-      val accountNumber = generateAccountNumberJson
-      handleRequest(nsiPayload, Created(accountNumber), description)
+      nsiPayload.nbaDetails match {
+        case Some(bankDetails) ⇒ getBankProfile(BankDetails(bankDetails.sortCode, bankDetails.accountNumber)) match {
+          case Profile(None, None) ⇒
+            logger.warn("Create Account request failed, no BARSResponse or CreateAccountResponse were returned")
+            InternalServerError
+          case Profile(Some(bars), None) ⇒ if (bars.accountNumberWithSortCodeIsValid === true && bars.sortCodeIsPresentOnEISCD === "yes") {
+            logger.info("Create Account request made")
+            val accountNumber = generateAccountNumberJson
+            handleRequest(nsiPayload, Created(accountNumber), description)
+          } else {
+            logger.warn("Create Account request failed due to BARS validation")
```

Comment:
we don't want to care about the BARs response here - here, we're pretending to be NS&I so we're only interested in the create account response part of the profile

Reply:
ok thank you